src/query/src/part_sort.rs

```diff
@@ -131,6 +154,10 @@ impl ExecutionPlan for PartSortExec {
     fn metrics(&self) -> Option<MetricsSet> {
         Some(self.metrics.clone_inner())
     }
+
+    fn benefits_from_input_partitioning(&self) -> Vec<bool> {
```

Comment:
Why though, does set this to false prevent datafusion to do wrong optimization? Maybe a short comment here?